app/src/main/res/values-es/strings.xml

```diff
@@ -55,4 +55,39 @@
     <string name="last_month">Last month</string>
     <string name="last_year">Last year</string>
     <string name="all_time">Todo el tiempo</string>
+    <string name="graphs">Graphs</string>
+    <string name="graph_deficit">Deficit/surplus</string>
+    <string name="graph_length">Duration</string>
+    <string name="graph_start">Start time</string>
+    <string name="graph_rating">Rating</string>
+    <string name="graph_average">Average</string>
+    <string name="graph_total">Total</string>
+    <string name="graph_hours">Hours</string>
+    <string name="graph_ideal">Ideal</string>
+    <string name="settings_ideal_sleep_length">Ideal sleep length</string>
+    <string name="hours_2">"2 hours"</string>
```

Comment:
Do you think you could try to use the standard hour/minute picker instead? It would allow silly things like picking 2h45m and also would get rid of long strings to localize.

Reply:
I decided to do it this way for a few reasons:

1. While there is `TimePickerDialog`, there isn't a preference equivalent, so I'd need a bit more code to implement a preference version ([relevant StackOverflow](https://stackoverflow.com/questions/5533078/timepicker-in-preferencescreen), alternatively a library like https://github.com/koji-1009/ChronoDialogPreference).

2. To begin with, this value is not supposed to be a time, it's a duration. If the user is presented with the analog clock from `TimePickerDialog`, I figured it might be a bit unintuitive, especially to select durations longer than 12 hours.

For reference, Sleep as Android uses a slider + list preference:

<details>
<summary>Sleep as Android screenshots</summary>

![Screenshot_20210302-200408_1](https://user-images.githubusercontent.com/2847492/109751187-c70ea700-7b92-11eb-85ec-ff1224f39ca9.png)
![Screenshot_20210302-200434_1](https://user-images.githubusercontent.com/2847492/109751202-d130a580-7b92-11eb-8da1-9bc3a5de1dac.png)

</details>

I originally tried doing something similar, but even with `SeekBarPreference` there's some customization needed (showing just 480 minutes or whatever isn't too user friendly), so I got lazy and stuck with just the list.

Another alternative is importing another library, eg https://github.com/svenwiegand/time-duration-picker, though that isn't maintained anymore.

Given all of this do you have a preference?